jmclient/jmclient/taker.py

```diff
@@ -483,7 +487,8 @@ def receive_utxos(self, ioauth_data):
             num_ins = len([u for u in sum(self.utxos.values(), [])])
             num_outs = len(self.outputs) + 1
             new_total_fee = estimate_tx_fee(num_ins, num_outs,
-                                    txtype=self.wallet_service.get_txtype())
+                                    txtype=self.wallet_service.get_txtype(),
+                                    outtype=self.wallet_service.get_outtype(self.coinjoin_address()))
```

Comment:
And the same applies here, i.e. https://github.com/JoinMarket-Org/joinmarket-clientserver/pull/921/files#r701344743

Reply:
A refinement to that statement, though.

As discussed in the first response to the PR, we have two very different contexts in which we call `estimate_tx_fee`, with the first genuinely being an *estimate* sometimes very rough, and the second being something close to an exact calculation, because the inputs and outputs are known. This is the case, here.

I see two obvious approaches: just accept the minor inaccuracy if one of the outputs is not the wallet type (that's what we had before, and as just mentioned, it is very rare that the type is different), or, have this new `outtype` argument be a list or tuple, with one for each output, so we can account exactly, in case the output types are different. But it's a hassle and I'm not sure it's worth it here.

Reply:
You're right, if we are to be exact, we should use a sequence for every output (and input) in order to be able to accurately determine the fees and not just an estimate. This modification was made based on the unlikely chance that a single output type would be different from the transaction type itself. The reason why these modifications were made was to provide a more accurate estimate of the fees when the very lowest fee rate (1 sat/vB) was specified. The minute difference in the estimate can be tolerated when using higher fee rates so this should really be an improvement for those who use the lowest fee rate.

I was trying to make as little a modification as possible but we should consider making it handle more scenarios? Especially now that we have fidelity bonds. That would require making significant changes to the code base.

Reply:
I wouldn't worry about fidelity bonds.

My main motivation for commenting here was based on the error as per previous comments.